src/core/radio/trel_interface.cpp

```diff
@@ -383,6 +383,12 @@ Error Interface::Send(const Packet &aPacket, bool aIsDiscovery)
     }
 
 exit:
+    ++mTrelCounters.mTxPackets;
+    mTrelCounters.mTxBytes += aPacket.GetLength();
+    if (error != kErrorNone)
+    {
+        ++mTrelCounters.mTxFailure;
```

Comment:
agree.

Maybe not this time, here another thought is we may want to count the actual traffic to infra link by TREL, for above broadcast case, there may be multiple send operations

Reply:
The whole trel_interface.cpp/hpp file is guarded by OPENTHREAD_CONFIG_RADIO_LINK_TREL_ENABLE flag. So all logic in the interface will only be build when TREL is enabled.

Reply:
Trel can be still disabled at run time (or through `otTrelSetFilterEnabled`) when `RADIO_LINK_TREL_ENABLE` is in use.

> Maybe not this time, here another thought is we may want to count the actual traffic to infra link by TREL, for above broadcast case, there may be multiple send operations

I agree with @librasungirl. I think would be good to add this in this PR itself. Broadcast messages are send as a bunch of unicast message to the peers, so counting this as one does not sound correct to me.

Here is my suggestion on how to add this:
- Can define a new method `SendPacket()` with similar input parameters as `otPlatTrelSend()` 
- Change this method to call `SendPacket()` instead of `otPlatTrelSend()`.
- The new `SendPackte()` can then update the counter and call `otPlatTrelSend()`.
- This way the counters would correctly get updated for all traffic going over the TREL interface (particualr when bordacast message are sent as bunch of unciast).